dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSBenchmarkTest.java

```diff
@@ -0,0 +1,366 @@
+ *******************************************************************************/
+package org.hpccsystems.dfs.client;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import static org.junit.Assert.fail;
+
+import org.hpccsystems.dfs.client.HPCCFile;
+import org.hpccsystems.dfs.client.HPCCRecord;
+import org.hpccsystems.dfs.client.HPCCRecordBuilder;
+import org.hpccsystems.dfs.client.HpccRemoteFileReader;
+import org.hpccsystems.dfs.client.DataPartition;
+
+import org.hpccsystems.dfs.cluster.*;
+import org.hpccsystems.commons.ecl.FieldDef;
+import org.hpccsystems.commons.errors.HpccFileException;
+import org.hpccsystems.ws.client.platform.test.BaseRemoteTest;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(RemoteTests.class)
+public class DFSBenchmarkTest extends BaseRemoteTest
+{
+    private static final String[] datasets = {"benchmark::integer::100mb","benchmark::string::100mb","benchmark::varstring::100mb","benchmark::utf8::100mb",
+                                            "benchmark::unicode::100mb","benchmark::real::100mb","benchmark::decimal::100mb"};
+    private static final float[] minDatasetConstructionEfficiency = {0.90f,0.90f,0.90f,0.90f,0.90f,0.90f,0.90f};
+
+    @Before
+    public void setup() throws Exception
+    {
+        super.setup();
+    }
+
+    @Test
+    public void integrationReadBenchmark() throws Exception
+    {
+        int numReadSamples = 5;
+        for (int i = 0; i < datasets.length; i++)
+        {
+            HPCCFile file = new HPCCFile(datasets[i], connString , hpccUser, hpccPass);
+            file.setFileAccessExpirySecs(1000);
+
+            // Raw streaming
+            ArrayList<Float> rawReadTimes = new ArrayList<Float>(numReadSamples);
+            for (int j = 0; j < numReadSamples; j++)
+            {
+                long startTime = System.currentTimeMillis();
+                try
+                {
+                    readRawFileData(file);
+                }
+                catch(Exception e)
+                {
+                    Assert.fail(e.getMessage());
+                }
+
+                long endTime = System.currentTimeMillis();
+
+                float timeInSeconds = (endTime - startTime);
+                timeInSeconds /= 1000.0;
+                rawReadTimes.add(new Float(timeInSeconds));
+            }
+
+            // Remove longest sample (Remove long samples due to wwarmup)
+            Float longestSample = rawReadTimes.get(0);
+            int sampleToRemove = 0;
+            for (int j = 0; j < rawReadTimes.size(); j++) {
+                Float currentReadSample = rawReadTimes.get(j);
+                if (longestSample < currentReadSample) {
+                    longestSample = currentReadSample;
+                    sampleToRemove = j;
+                }
+            }
+            rawReadTimes.remove(sampleToRemove);
```

Comment:
It might be worthwhile to log this action along w/ the info about the value being removed